### PR TITLE
Fix horse stats being re-rolled on login

### PR DESCRIPTION
**Pull request by MrJeremyFisher**
Fixes logging out then back in rerolling horse stats on both untamed and tamed horses.

### CODE REVIEW

**Review by okx-code — COMMENTED** [review]
Would using [CreatureSpawnEvent.SpawnReason.SPAWNER_EGG](https://hub.spigotmc.org/javadocs/spigot/org/bukkit/event/entity/CreatureSpawnEvent.SpawnReason.html#SPAWNER_EGG) and [CreatureSpawnEvent.SpawnReason.NATURAL](https://hub.spigotmc.org/javadocs/spigot/org/bukkit/event/entity/CreatureSpawnEvent.SpawnReason.html#NATURAL) work instead?

File: plugins/simpleadminhacks-paper/src/main/java/com/programmerdan/minecraft/simpleadminhacks/hacks/basic/FasterHorses.java [hunk 1]
```diff
@@ -39,25 +46,30 @@ public void on(EntityBreedEvent event) {
...
 		event.getEntity().getAttribute(Attribute.GENERIC_MOVEMENT_SPEED).setBaseValue(newSpeed);
-
-		plugin.getLogger().log(Level.INFO, "Horse breed to have speed: " + newSpeed);
+		event.getEntity().getPersistentDataContainer().set(speedChangedKey, PersistentDataType.INTEGER, 1);
+		plugin.getLogger().log(Level.INFO, "Horse bred to have speed: " + newSpeed);
 	}
 
 	@EventHandler(priority = EventPriority.HIGHEST)
 	public void on(CreatureSpawnEvent event) {
+		System.out.println("evt");
```
Comment (AngrySoundTech) [inline plugins/simpleadminhacks-paper/src/main/java/com/programmerdan/minecraft/simpleadminhacks/hacks/basic/FasterHorses.java:55]: Debug print

**Review by AngrySoundTech — CHANGES_REQUESTED** [review]

**Comment by MrJeremyFisher** [comment]
The Natural spawn reason is called when the server loads an entity and when an entity is spawned, so no.

**Comment by okx-code** [comment]
ok, what about [getTicksLived](https://hub.spigotmc.org/javadocs/spigot/org/bukkit/entity/Entity.html#getTicksLived())?

**Comment by MrJeremyFisher** [comment]
Because the horse is spawned every time the riding player logs out and back in, its age at that instant is 0. At least that's what I saw when I tried that as an option

File: plugins/simpleadminhacks-paper/src/main/java/com/programmerdan/minecraft/simpleadminhacks/hacks/basic/FasterHorses.java [hunk 2]
```diff
@@ -29,6 +35,9 @@ public void on(EntityBreedEvent event) {
 		if (event.getEntity().getType() != EntityType.HORSE) {
 			return;
 		}
+		if (event.getEntity().getPersistentDataContainer().has(speedChangedKey, PersistentDataTypes.BOOLEAN)) {
+			return;
+		}
```
Comment (okx-code) [inline plugins/simpleadminhacks-paper/src/main/java/com/programmerdan/minecraft/simpleadminhacks/hacks/basic/FasterHorses.java:40]: This is not necessary and can be removed

**Review by okx-code — APPROVED** [review]

**Review by RedDevel2 — APPROVED** [review]

**Pull request merged**